test/extended/oauth/server_headers.go

```diff
@@ -0,0 +1,121 @@
+package oauth
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	t "github.com/onsi/ginkgo/extensions/table"
+	o "github.com/onsi/gomega"
+
+	"k8s.io/client-go/rest"
+	"k8s.io/kube-openapi/pkg/util/sets"
+
+	osinv1 "github.com/openshift/api/osin/v1"
+
+	"github.com/openshift/origin/test/extended/util"
+	"github.com/openshift/origin/test/extended/util/oauthserver"
+)
+
+var _ = g.Describe("[Feature:OAuthServer] [Headers]", func() {
+	var (
+		transport       http.RoundTripper
+		oauthServerAddr string
+		oc              *util.CLI
+	)
+	oc = util.NewCLI("oauth-server-headers", util.KubeConfigPath())
+
+	g.BeforeEach(func() {
+		var err error
+
+		transport, err = rest.TransportFor(rest.AnonymousClientConfig(oc.UserConfig()))
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		// deploy oauth server
+		osinProvider, err := oauthserver.GetRawExtensionForOsinProvider(&osinv1.AllowAllPasswordIdentityProvider{})
```

Comment:
While it does not matter that much for this exact test, the `AllowAll` IdP is not really supported in 4.x anymore, do you think you could set up a simple HTPasswd IdP instead?